### PR TITLE
dedup shard IDs when picking local partitions

### PR DESCRIPTION
**Pull request by jerry-stripe**
**Summary**
Only consider unique shard IDs when choosing partition assignments.

**Motivation**
This will allow nodes to share partition assignments by using the same shard ID such that one node can replace another in the case of failure.

r? @vasi-stripe 
cc? @stripe/storage

### CODE REVIEW

**Comment by vasi-stripe** [comment]
This looks good! Let's test it out in QA before we merge :)

File: status.go [hunk 1]
```diff
@@ -432,7 +432,6 @@ func (vs *version) status() versionStatus {
 		CreatedAt:  vs.created.UTC().Truncate(time.Second),
 		State:      vs.state,
 		Partitions: partitions,
-		ShardID:    vs.sequins.config.Sharding.ShardID,
```
Comment (jerry-stripe) [inline status.go]: Probably should have put this in another PR, but I forgot to flag this in yours 😅 

This misses a lot of cases where the ShardID can be set (self-assigned id file, smallest missing, etc) and the Shard ID is already exposed in the header of requests so I feel like this isn't needed. Unless you explicitly wanted to expose what was set in the config file?

**Comment by jerry-stripe** [comment]
@vasi-stripe Deployed to standardsequins in QA and everything looks peachy 👍

**Comment by vasi-stripe** [comment]
lgtm
